src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -11,21 +11,20 @@ internal class RemoteJSRuntime : JSRuntimeBase
     {
         private IClientProxy _clientProxy;
 
-        public RemoteJSRuntime()
-        {
-        }
-
-        internal void Initialize(IClientProxy clientProxy)
+        internal void Initialize(CircuitClientProxy clientProxy)
         {
             _clientProxy = clientProxy ?? throw new ArgumentNullException(nameof(clientProxy));
         }
 
         protected override void BeginInvokeJS(long asyncHandle, string identifier, string argsJson)
         {
-            if (_clientProxy == null)
+            if (_clientProxy == CircuitClientProxy.OfflineClient)
             {
-                throw new InvalidOperationException("The JavaScript runtime is not available during prerendering.");
+                var errorMessage = "JavaScript interop calls cannot be issued while the client is not connected, because the server is not able to interop with the browser at this time. " +
+                    "Components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during periods where the client is not connected.";
+                throw new InvalidOperationException(errorMessage);
```

Comment:
> Components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during periods where the client is not connected

Do we actually expose this?

Reply:
Not as yet. Suggestions on where it might go? `Circuit.IsClientConnected`?

Reply:
We don't have a property for this (we might have to expose one) but this should at least be in a try catch

Reply:
We could expose whether we are connected or not
https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events
We'll need some sort of abstraction for this that tells you whether you are online or offline in WebAssembly this can just be backed by the appropriate browser API. In remote environments we shim it.

Reply:
> but this should at least be in a try catch

That would be anytime they call `JSRuntime`, no? That feels super ugly. Plus offline in WebAssembly has a very different meaning than in Components - in the former the client cannot connect to the server so your `HttpClient` calls fail, versus here your component methods throw. Would we want to conflate the two?